rfc#0001

```diff
@@ -0,0 +1,140 @@
+of the workload, either for reloading ("recycling" a Keep) or to
+transfer to another Keep in the same trust domain.  This caching, if
+under the origincal WPEK, is at the expense of Perfect Forward
+Secrecy.  If a Keep is to transfer the workload to another Keep in
+the same domain, it MAY encrypt it under a new WPEK.  If the
+original is used, it will need to established in the new Keep either
+at provisioning time (e.g. by the ECA) or at a later date through
+key negotiation.  Such functionality is considered risky, as the key
+management required is difficult to get right, and provides new
+options for compromise or design/implementation errors.  Any design
+of such functionality requires a new RFC (see "Future possibilities"
+below).
+
+The Keep MUST NOT accept unencrypted workloads, as there is no way
+to verify their provenance.
+The Keep SHOULD listen on a well-known port (see "Unresolved
+questions").
+If a port has been pre-esablished, the Keep MUST listen on this port,
+and MAY also listen on the well-known port.
+The Keep MAY cache the workload for re-use, or transfer to another
+Keep.  If a Keep allows workloads to be used by another Keep, the
+workload MAY be encrypted under a new WPEK, established between the
+original Keep and new Keep.
+Keeps MAY be implemented to enable the capability to performing
+workload caching.
+Enabling of the capability for a Keep to perform caching MUST be at
+the discretion of the ECA or entities within the ECA's trust domain.
+Enabling of the capability for a Keep to perform caching MUST NOT
+be at the discretion of the host.  The host MUST NOT be able to
+enable caching.
+
+## Reference
+
+Empty.
+
+## Drawbacks
+
+Why should we *not* do this?
+
+## Rationale and alternatives
+
+Alternatives considered:
+
+Enarx client agent encrypts workload, sends to CPU/firmware via Enarx
+host agent. CPU/firmware decrypts workload, loads into Keep. This has
+the advantage that we use an existing network connection, and there's
+no need to establish a connection nor a session key between the Keep
+and the client agent.
+Rejected: this is not supported by all architectures.
+
+Enarx client agent encrypts workload, sends to host agent. Host agent
+sends to Keep via a listening agent in the Keep. Keep decrypts, loads
+and runs. This maintains the host agent's status as a proxy between
+other components, and doesn't make further assumptions about routing
+which is made by 2. 
+Rejected:
+- the host agent should not need logic to provide functionality.
+- maintaining multiple versions is complex.
+
+## Prior art
+
+Some TEE implementations assume trust in the host system.  This is
+explicitly excluded as one of the Enarx project's design principles.
+In such implementations, a component trusted by the host system may
+handle key management and decryption/loading of the workload.
+
+## Unresolved questions
+
+1. What should the well-known port number be?  Should we use HTTPS
```

Comment:
I suggest to use a port >1024 so that we can bind the ports on the host as a non-root user. Which one specifically doesn't seem important, and any unassigned by IANA will do.

Reply:
Except that HTTPS is on 443.  Certainly requires more discussion.

Reply:
I don't know of any plans to use HTTPS, specifically on its default port. Maybe it can be left out of the RFC, and _well-known communication endpoint_ might suffice to specify the behavior. See https://github.com/enarx/rfcs/pull/2#discussion_r377204173

Reply:
This is here as a discussion point, and we may well remove it.  Given text elsewhere about _not_ encrypting at the connection level, HTTPS may not make sense.

Reply:
Changed this from HTTPS, but left it with "port number" see notes above.